documentation/book/con-strategies-for-upgrading-clients.adoc

```diff
@@ -0,0 +1,65 @@
+// Module included in the following assemblies:
+//
+// assembly-upgrading-kafka-versions.adoc
+
+[id='con-strategies-for-upgrading-clients-{context}']
+
+= Strategies for upgrading clients
+
+The best approach to upgrading your client applications (including Kafka Connect connectors) depends on your particular circumstances.
+
+Consuming applications need to receive messages in a message format that they understand. You can ensure that this is the case in one of two ways:
+
+* By upgrading all the consumers for a topic _before_ upgrading any of the producers.
+* By having the brokers down-convert messages to an older format.
+
+Using broker down-conversion puts extra load on the brokers, so it is not ideal to rely on down-conversion for all topics for a prolonged period of time.
+For brokers to perform optimally they should not be down converting messages at all.
+
+Broker down-conversion is configured in two ways:
+
+* The topic-level `message.format.version` configures it for a single topic.
+
+* The broker-level `log.message.format.version` is the default for topics that do not have the topic-level `message.format.version` configured.
+
+Another aspect to consider is that once new-version messages have been published to a topic then they will be visible to consumers.
```

Comment:
I'm not sure I follow what this is saying.  Consider rewording.

Reply:
Changed to: 'Messages published to a topic in a new-version format will be visible to consumers, because brokers perform down-conversion when they receive messages from producers, not when they are sent to consumers.'